relay-server/src/metrics_extraction/transactions/mod.rs

```diff
@@ -543,6 +542,33 @@ fn extract_span_metrics(
     Ok(())
 }
 
+fn normalize_domain(domain: &str, port: Option<&String>) -> Option<String> {
+    let tokens = domain.split('.').collect::<Vec<&str>>();
+    if tokens.is_empty() {
+        return None;
+    }
+
+    if tokens.len() == 1 {
+        if let Some(p) = port {
+            return Some(format!("{}:{}", tokens[0], p));
+        }
+        return Some(tokens[0].to_owned());
+    }
+
+    // sth.subdomain.domain.tld => *.domain.tld
+    // TODO(iker): find a proper way to do
+    // this and consider cloud domains, e.g.
+    // <name>.service.<cloud-region>.<whatever>
+    let extracted_domain: Vec<&&str> = tokens.iter().rev().take(2).rev().collect();
+    if let Some(p) = port {
+        return Some(format!(
+            "*.{}.{}:{}",
+            extracted_domain[0], extracted_domain[1], p
+        ));
+    }
+    Some(format!("*.{}.{}", extracted_domain[0], extracted_domain[1]))
```

Comment:
What do you think of this simplification?
```suggestion
    let mut parts = domain.rsplitn(3, '.');
    let tld = parts.next();
    let domain = parts.next();
    let prefix = parts.next().map(|_| "*");

    let mut replaced = prefix
        .iter()
        .chain(domain.iter())
        .chain(tld.iter())
        .join(".");
    if let Some(port) = port {
        replaced = format!("{}:{}", replaced, port);
    }
    Some(replaced)
```